src/ameba/cli/cmd.cr

```diff
@@ -177,6 +182,30 @@ module Ameba::Cli
     config
   end
 
+  private def root_path_from_opts(opts)
+    return unless globs = opts.globs
+    root =
+      case
+      when path = opts.config
+        path.parent
+      when path = globs.find(&->File.file?(String))
+        Path[path].parents.reverse!.find(&->root_path?(Path))
+      when path = globs.find(&->File.directory?(String))
+        path = Path[path]
+        if root_path?(path)
+          path
+        else
+          path.parents.reverse!.find(&->root_path?(Path))
+        end
+      end
```

Comment:
why do we check only the first file/directory (using `find` method) if globs could have multiple?

Reply:
How else would you do it?